examples/kops-kube-vpc/Makefile

```diff
@@ -0,0 +1,80 @@
+# Hardcoding value of 3 minutes when we check if the plan file is stale
+STALE_PLAN_FILE := `find "tf.out" -mmin -3 | grep -q tf.out`
+SHELL := /bin/bash
+
+AWS_AVAILABILITY_ZONE := `terraform output -json | jq '.availability_zone.value' | tr -d '"'`
+KOPS_STATE_BUCKET := `terraform output -json | jq '.state_bucket_id.value' | tr -d '"'`
+KOPS_CLUSTER_NAME := `terraform output -json | jq '.subdomain_zone_name.value' | tr -d '"'`
+KOPS_CLUSTER_ZONE := `terraform output -json | jq '.subdomain_zone_id.value' | tr -d '"'`
+KOPS_AWS_VPC := `terraform output -json | jq '.vpc_id.value' | tr -d '"'`
```

Comment:
I'd really like it if these were explicitly set in the example/env file rather than being semi automatic like this. The reason being that the tf code will live in the terraform directory and the kubernetes code will live in another directory with it's separate Makefile and we'd like to avoid to semi magic implicit variables like this. Does that make sense in the case of this example ?

Reply:
> Does that make sense in the case of this example ?

I guess I went overboard with DRY. Although this example is TF + Kops, I think, now that you mention this (and on Slack too), that coupling the two this tightly could be less helpful than imagined.

Reply:
Here are some more concrete thoughts on how to avoid the copying-and-pasting of values:

- Add a top-level `Makefile` that is in charge of coordinating between the Terraform and the Kubernetes

    - Add a target that writes `kubernetes/<ENVIRONMENT>/tf_env` containing things like the VPC_ID, VPC_CIDR, SUBNET_IDS, SUBNET_CIDRS, and anything else that can be gotten from the `terraform/<ENVIRONMENT>/env` or `terraform output`s.  BTW, you can get the value from a Terraform output using `terraform output <OUTPUT_NAME>` rather than using `jq`.

    - Also add a target that copies the SSH key

- Modify `kubernetes/<ENVIRONMENT>/env` to `source "$(dirname "${BASH_SOURCE[0]}")/tf_env"` and then set any values that aren't in the tf-env.  There probably won't be much left besides the `KOPS_MASTER_*`, `NODE_*`, and `KUBE*` variables.

- In `kubernetes/Makefile`, add a target that uses `kops get cluster fpco-prod-v2.fpcomplete.com -o json`, `jq`, and `kops replace -f` to update the subnet IDs and CIDRs from the env.  This would replace manually editing the cluster manifest.

Thoughts?